lib/shared/config/cli-flags.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var yargs = require('yargs');
+var camelCase = require('lodash.camelcase');
+
+function mergeToCliFlags(cliFlags, config, cliOptions) {
+  var argv = yargs(process.argv.slice(2)).argv;
+  var cfgFlags = config.flags || {};
+
+  var optNames = ['silent'];
+  optNames.filter(excludeArgsByUser).map(camelCase).forEach(copyConfig);
+
+  function excludeArgsByUser(name) {
```

Comment:
For example, when both `--gulpfile` command line option and `flags.gulpfile` in a config file are specified, command line option should be used rather than config file flag, I think.

But merging config file flags into `opts` and `env` simply, `opts.gulpfile` and `env.modulePath` specified by `--gulpfile` are overridden by `flags.gulpfile` from a config file.

So before merging, it is needed to get rid of `opts.gulpfile` and `env.modulePath` from targets to be merged. (It is also good to get rid of `flags.gulpfile` from config file flags marging.)